crowbar_framework/app/models/proposal.rb

```diff
@@ -68,6 +68,13 @@ def increment_crowbar_revision!
     else
       item["deployment"][self.barclamp]["crowbar-revision"] += 1
     end
+    # The domain attribute of the dns barclamp is needed when looking up a Node.
+    # We don't want to go to the database every time we need it since that will
+    # be many times per second, but nor can we statically record it since it may
+    # change. Instead we use Rails' caching mechanism to store the new revision,
+    # and when we look it up elsewhere we first check the revision to see if the
+    # barclamp has been updated and we require a trip to the database.
+    cache_crowbar_revision if barclamp == 'dns'
```

Comment:
or cache the value for every barclamp? `cache_crowbar_revision` looks ready for that